src/main/service.js

```diff
@@ -102,6 +102,24 @@ function changePassword(req, res, next) {
         });
 }
 
+/**
+ * reset user password
+ * @param req.username - the user's username
+ * @returns {Error} - if there is no registered user with this username
+ */
+function resetPassword(req, res, next) {
+    application.resetPassword(req.body).then(
+        businessUser => {
```

Comment:
the method doesn't return a 'businessUser'. change this parameter to `result` or simply `r` to indicate successful completion